pymeasure/experiment/parameters.py

```diff
@@ -468,92 +472,165 @@ def __repr__(self):
+    :param \\**kwargs: keyword-arguments passed to :class:`.Parameter`
+    """
+    pass
 
-    @value.setter
-    def value(self, value):
-        self._value = value
 
+class BooleanParameter(Parameter, BooleanInputField):
+    """ :class:`.Parameter` subclass that uses the boolean type to store the value.
 
-class Metadata(object):
+    :var value: The boolean value of the parameter
+
+    Refer to :class:`.Parameter` for the initialization arguments and keyword-arguments.
+    """
+    pass
+
+
+class FloatParameter(Parameter, FloatInputField):
+    """ :class:`.Parameter` subclass that uses the floating point type to store the value.
+
+    :var value: The floating point value of the parameter
+
+    :param name: the parameter name
+    :param units: The units of measure for the parameter
+    :param minimum: The minimum allowed value (default: -1e9)
+    :param maximum: The maximum allowed value (default: 1e9)
+    :param decimals: The number of decimals considered (default: 15)
+    :param step: step size for the field's UI spinbox. If None, spinbox will have step disabled
+    :param \\**kwargs: keyword-arguments passed to :class:`.Parameter`
+    """
+    pass
+
+
+class VectorParameter(Parameter, VectorInputField):
+    """ :class:`.Parameter` subclass that stores the value in a vector format.
+
+    :var value: The value of the parameter as a list of floating point numbers
+
+    :param name: the parameter name
+    :param length: The integer dimensions of the vector
+    :param units: The units of measure for the parameter
+    :param \\**kwargs: keyword-arguments passed to :class:`.Parameter`
+    """
+    pass
+
+
+class ListParameter(Parameter, ListInputField):
+    """ :class:`.Parameter` subclass that stores the value as a list.
+    String representation of choices must be unique.
+
+    :param name: the parameter name
+    :param choices: An explicit list of choices, which is disregarded if None
+    :param units: The units of measure for the parameter
+    :param \\**kwargs: keyword-arguments passed to :class:`.Parameter`
+    """
+    pass
+
+
+class PhysicalParameter(Parameter, PhysicalInputField):
+    """ :class:`.VectorParameter` subclass of 2 dimensions to store a value
+    and its uncertainty.
+
+    :var value: The value of the parameter as a list of 2 floating point numbers
+
+    :param name: the parameter name
+    :param uncertainty_type: Type of uncertainty, 'absolute', 'relative' or 'percentage'
+    :param units: The units of measure for the parameter
+    :param \\**kwargs: keyword-arguments passed to :class:`.Parameter`
+    """
+    pass
+
+
+class Metadata(InputField):
     """ Encapsulates the information for metadata of the experiment with
     information about the name, the fget function and the units, if supplied.
     If no fget function is specified, the value property will return the
     latest set value of the parameter (or default if never set).
+    Subclass of :class:`.InputField`
```

Comment:
Is it necessary to state the parent class?

In the api documentation you can toggle "show_inheritance" to show that information.

That makes the actual inheritance the single point of truth.